packages/react-relay/__tests__/ClientEdges-test.js

```diff
@@ -31,362 +37,377 @@ injectPromisePolyfill__DEPRECATED();
+      function TestComponent() {
+        return (
+          <RelayEnvironmentProvider environment={environment}>
+            <React.Suspense fallback="Loading">
+              <InnerComponent />
+            </React.Suspense>
+          </RelayEnvironmentProvider>
+        );
+      }
+      const variables = {return_null: true};
+      function InnerComponent() {
+        const data = useLazyLoadQuery(
+          graphql`
+            query ClientEdgesTest4Query($return_null: Boolean!) {
+              me {
+                client_object(return_null: $return_null)
+                  @required(action: THROW) {
+                  description
+                }
+              }
             }
-          }
-        }
-      `,
-      variables,
-    );
+          `,
+          variables,
+        );
 
-    return data.me?.client_object?.description;
-  }
-  expect(() => {
-    TestRenderer.act(() => {
-      TestRenderer.create(<TestComponent />);
+        return data.me?.client_object?.description;
+      }
+      expect(() => {
+        TestRenderer.act(() => {
+          TestRenderer.create(<TestComponent />);
+        });
+      }).toThrow(
+        "Relay: Missing @required value at path 'me.client_object' in 'ClientEdgesTest4Query'.",
+      );
+      expect(fetchFn.mock.calls.length).toEqual(0);
     });
-  }).toThrow(
-    "Relay: Missing @required value at path 'me.client_object' in 'ClientEdgesTest4Query'.",
-  );
-  expect(fetchFn.mock.calls.length).toEqual(0);
-});
-
-/**
- * @RelayResolver User.same_user_client_edge: User
- */
-export function same_user_client_edge(): {id: string} {
-  return {id: '1'};
-}
-
-// https://github.com/facebook/relay/issues/4882
-it('should fetch data missing in fragment spread within `@waterfall` field', () => {
-  function TestComponent() {
-    return (
-      <RelayEnvironmentProvider environment={environment}>
-        <React.Suspense fallback="Loading">
-          <InnerComponent />
-        </React.Suspense>
-      </RelayEnvironmentProvider>
-    );
-  }
 
-  const variables = {};
-  function InnerComponent() {
-    const data = useLazyLoadQuery(
-      graphql`
-        query ClientEdgesTest5Query {
-          me {
-            same_user_client_edge @waterfall {
-              # No fields here means that we render without detecting any
-              # missing data here and don't attempt to fetch the @waterfall
-              # query.
-              #
-              # The same bug can be tirggered by adding a field that is already
-              # in the store for an unrelated reason.
-              ...ClientEdgesTest5Query_user
-              # Adding "name" here will cause the query to be fetched.
-            }
-          }
-        }
-      `,
-      variables,
-    );
+    // https://github.com/facebook/relay/issues/4882
+    it.each([[true], [false]])(
+      'should fetch data missing in fragment spread within `@waterfall` field. CHECK_ALL_FRAGMENTS_FOR_MISSING_CLIENT_EDGES = %s',
```

Comment:
@jonreading81 any chance you could create a repro unit test similar to this demonstrating the resolver issue you are seeing? If not, just providing more details might be able to help me construct one.